### PR TITLE
Update PyTorch surrogate example

### PR DESCRIPTION
**Pull request by RTSandberg**
This PR updates the PyTorch surrogate model example from PR #482 to reflect work done for the PASC revision.
It also improves on the original presentation. 

New figures here:
Schematic:
schema_15_stages

Moments:
lpa_ml_surrogate_moments

Initial phase space:
initial_phase_spaces

Final phase space:
stage_14_phase_spaces

### CODE REVIEW

**Review by ax3l — COMMENTED** [review]
Thanks that's awesome!

Do you want to:
- update the `.. figure::` URLs,
- remove the 9 stage files?

**Comment by RTSandberg** [comment]
Yes, I just updated the figures and cleared out the 9 stage files

File: examples/pytorch_surrogate_model/README.rst [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
 
 A schematic with more information can be seen in the figure below:
 
-.. figure:: https://user-images.githubusercontent.com/10621396/289956389-c7463b99-fb56-490a-8511-22c43f45cdf8.png
-   :alt: [fig:lpa_schematic] Schematic of the 9 stages of laser-plasma accelerators.
+.. figure:: https://gist.githubusercontent.com/RTSandberg/cf3f6193b3da12e7fd815f69789fd6f2/raw/2308e412d7482d55811afa2e9a0c6fa97627fc2f/schema_15_stages.png
+   :alt: [fig:lpa_schematic] Schematic of the 15 stages of laser-plasma accelerators.
 
-   [fig:lpa_schematic] Schematic of the 9 stages of laser-plasma accelerators.
+   [fig:lpa_schematic] Schematic of the 15 stages of laser-plasma accelerators.
```
Comment (ax3l) [inline examples/pytorch_surrogate_model/README.rst]: Shows an unused label in the rendered version:
```suggestion
   Schematic of the 15 stages of laser-plasma accelerators.
```

File: examples/pytorch_surrogate_model/README.rst [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
   **Hybrid Beamline Element ML-Training for Surrogates in the ImpactX Beam-Dynamics Code**.
   14th International Particle Accelerator Conference (IPAC'23), WEPA101, 2023.
   `DOI:10.18429/JACoW-IPAC2023-WEPA101 <https://doi.org/10.18429/JACoW-IPAC2023-WEPA101>`__
 
 A schematic with more information can be seen in the figure below:
 
-.. figure:: https://user-images.githubusercontent.com/10621396/289956389-c7463b99-fb56-490a-8511-22c43f45cdf8.png
-   :alt: [fig:lpa_schematic] Schematic of the 9 stages of laser-plasma accelerators.
+.. figure:: https://gist.githubusercontent.com/RTSandberg/cf3f6193b3da12e7fd815f69789fd6f2/raw/2308e412d7482d55811afa2e9a0c6fa97627fc2f/schema_15_stages.png
+   :alt: [fig:lpa_schematic] Schematic of the 15 stages of laser-plasma accelerators.
```
Comment (ax3l) [inline examples/pytorch_surrogate_model/README.rst]:
```suggestion
   :alt: Schematic of the 15 stages of laser-plasma accelerators.
```

**Review by ax3l — COMMENTED** [review]
This looks excellent and like it has all the updates we added for device code in the revision :tada:

File: examples/pytorch_surrogate_model/README.rst (hunk 1 above)
Comment (RTSandberg) [reply examples/pytorch_surrogate_model/README.rst]: Yeah the figure references stopped working, I'm not sure why

**Review by ax3l — APPROVED** [review]
Thank you! :rocket: :sparkles:

**Pull request merged**